tests/ops/op_math/test_controlled.py

```diff
@@ -39,6 +39,32 @@
 # pylint: disable=pointless-statement
 # pylint: disable=expression-not-assigned
 
+
+def equal_if_decomposed(lhs, rhs):
```

Comment:
Could you clarify why this is needed? Does `qml.equal` fail when it's not supposed to?

Reply:
`qml.equal` is too picky and thinks `qml.MultiControlledX([0, 1], 2)` and `qml.Toffoli([0, 1, 2])` are different (there are also some other cases)